src/node_watchdog.cc

```diff
@@ -107,11 +115,92 @@ SigintWatchdog::~SigintWatchdog() {
 }
 
 
-void SigintWatchdog::HandleSigint() {
+bool SigintWatchdog::HandleSigint() {
   *received_signal_ = true;
   isolate_->TerminateExecution();
+  return /** stop propagation */true;
```

Comment:
If you want to be explicit here about the effects of the return value, you can use a two-valued `enum` instead of `bool`. (This is just fine too, of course.)